models/workitem2_repository.go

```diff
@@ -0,0 +1,118 @@
+	res := WorkItem{}
+	id, err := strconv.ParseUint(wi.ID, 10, 64)
+	if err != nil {
+		return nil, NotFoundError{entity: "work item", ID: wi.ID}
+	}
+
+	log.Printf("looking for id %d", id)
+	tx := r.db
+	if tx.First(&res, id).RecordNotFound() {
+		log.Printf("not found, res=%v", res)
+		return nil, NewNotFoundError("work item", wi.ID)
+	}
+
+	// Attributes is a string->string map hence need to add few conditions
+	var version int
+	// validate version attribute
+	if _, ok := wi.Attributes["version"]; ok {
+		versionStr := wi.Attributes["version"].(string)
+		version, err = strconv.Atoi(versionStr)
+		if err != nil {
+			return nil, NewBadParameterError("version", version)
+		}
+	} else {
+		return nil, VersionConflictError{simpleError{"version is mandatory"}}
+	}
+	if res.Version != version {
+		return nil, VersionConflictError{simpleError{"version conflict"}}
+	}
+
+	newWi := WorkItem{
+		ID:      id,
+		Type:    res.Type, // read WIT from DB object and not from payload relationship
+		Version: version + 1,
+		Fields:  res.Fields,
+	}
+
+	wiType, err := r.wir.LoadTypeFromDB(ctx, newWi.Type)
+	if err != nil {
+		// ideally should not reach this, if reach it means something went wrong while CREATE WI
+		return nil, NewBadParameterError("Type", newWi.Type)
+	}
+
+	rel := wi.Relationships
+	// TODO
+	// if rel.Assignee.Data == nil then remove the relationship for WI
+	if rel != nil && rel.Assignee != nil && rel.Assignee.Data != nil {
+		assigneeData := rel.Assignee.Data
+		identityRepo := account.NewIdentityRepository(r.db)
+		uuidStr := assigneeData.ID
+		if uuidStr == nil {
+			// remove Assignee
+			wi.Attributes[SystemAssignee] = nil
+		} else {
+			assigneeUUID, err := uuid.FromString(*uuidStr)
+			if err != nil {
+				return nil, NewBadParameterError("data.relationships.assignee.data.id", uuidStr)
+			}
+			_, err = identityRepo.Load(ctx, assigneeUUID)
+			if err != nil {
+				return nil, NewBadParameterError("data.relationships.assignee.data.id", uuidStr)
+			}
+			wi.Attributes[SystemAssignee] = *uuidStr
+			//  ToDO : make it a list and append
+			// existingAssignees := res.Fields[SystemAssignee]
+			// wi.Attributes.Fields[SystemAssignee] = append(existingAssignees, uuidStr)
+		}
+	}
+
+	for fieldName, fieldDef := range wiType.Fields {
+		fieldValue, exist := wi.Attributes[fieldName]
+		if !exist {
+			// skip non-mentioned Attributes because this is a PATCH request.
+			continue
+		}
+		var err error
+		newWi.Fields[fieldName], err = fieldDef.ConvertToModel(fieldName, fieldValue)
+		if err != nil {
+			return nil, NewBadParameterError(fieldName, fieldValue)
+		}
+	}
+
+	if err := tx.Save(&newWi).Error; err != nil {
+		log.Print(err.Error())
+		return nil, InternalError{simpleError{err.Error()}}
+	}
+	log.Printf("updated item to %v\n", newWi)
+	result, err := wiType.ConvertFromModel(newWi)
+	if err != nil {
+		return nil, InternalError{simpleError{err.Error()}}
```

Comment:
thanks, will keep in mind for upcoming work. Fixed in https://github.com/almighty/almighty-core/pull/482/commits/d8c08aea9e5f6cc075bed42bc55fb0972ec34018